src/common/hosts-patches-table.js

```diff
@@ -104,100 +114,70 @@ class HostsPatchesTable extends Component {
     return Promise.all(map(
       keys(pools),
       installAllPatchesOnPool
-    )).then(this._refreshMissingPatches)
-  }
-
-  _refreshHostMissingPatches = host => (
-    getHostMissingPatches(host).then(patches => {
-      this.setState({
-        missingPatches: {
-          ...this.state.missingPatches,
-          [host.id]: patches.length
-        }
-      })
-    })
-  )
-
-  _installAllHostPatches = host => (
-    installAllHostPatches(host).then(() =>
-      this._refreshHostMissingPatches(host)
-    )
-  )
-
-  componentWillMount () {
-    this._refreshMissingPatches()
+    ))
   }
 
   componentDidMount () {
     // Force one Portal refresh.
     // Because Portal cannot see the container reference at first rendering.
     this.forceUpdate()
+    this.unsubscribeMissingPatches = this._subscribeMissingPatches()
   }
 
   componentWillReceiveProps (nextProps) {
-    forEach(nextProps.hosts, host => {
-      const { id } = host
-
-      if (this.state.missingPatches[id] !== undefined) {
-        return
-      }
-
-      this.setState({
-        missingPatches: {
-          ...this.state.missingPatches,
-          [id]: 0
-        }
-      })
+    if (nextProps.hosts !== this.props.hosts) {
+      this.unsubscribeMissingPatches()
+      this.unsubscribeMissingPatches = this._subscribeMissingPatches(nextProps.hosts)
+    }
+  }
 
-      this._refreshHostMissingPatches(host)
-    })
+  componentWillUnmount () {
+    this.unsubscribeMissingPatches()
   }
 
   render () {
+    const {
+      buttonsGroupContainer,
+      container,
+      displayPools,
+      pools,
+      useTabButton
+    } = this.props
+
     const hosts = this._getHosts()
     const noPatches = isEmpty(hosts)
-    const { props } = this
 
-    const Container = props.container || 'div'
+    const Container = container || 'div'
 
-    const Button = this.props.useTabButton
+    const Button = useTabButton
       ? TabButton
       : ActionButton_
 
-    const Buttons = (
-      <Container>
-        <Button
-          handler={this._refreshMissingPatches}
-          icon='refresh'
-          labelId='checkForUpdates'
-        />
-        <Button
-          btnStyle='primary'
-          disabled={noPatches}
-          handler={this._installAllMissingPatches}
-          icon='host-patch-update'
-          labelId='installPoolPatches'
-        />
-      </Container>
-    )
-
     return (
       <div>
         {!noPatches
           ? (
             <SortedTable
               collection={hosts}
-              columns={props.displayPools ? POOLS_MISSING_PATCHES_COLUMNS : MISSING_PATCHES_COLUMNS}
+              columns={displayPools ? POOLS_MISSING_PATCHES_COLUMNS : MISSING_PATCHES_COLUMNS}
               userData={{
-                installAllHostPatches: this._installAllHostPatches,
+                installAllHostPatches,
```

Comment:
Is `this._installAllHostPatches` still used somewhere?
If not, have you correctly removed it?